libs/vkd3d/state.c

```diff
@@ -712,9 +807,11 @@ static HRESULT d3d12_root_signature_init_static_samplers(struct d3d12_root_signa
 static HRESULT d3d12_root_signature_init(struct d3d12_root_signature *root_signature,
         struct d3d12_device *device, const D3D12_ROOT_SIGNATURE_DESC *desc)
 {
+    const struct vkd3d_bindless_state *bindless_state = &device->bindless_state;
+    VkDescriptorSetLayout set_layouts[3 + VKD3D_MAX_BINDLESS_DESCRIPTOR_SETS];
```

Comment:
If VKD3D_MAX_BINDLESS_DESCRIPTOR_SETS is 6, wouldn't this overflow the limit of 8?

Reply:
This is really just here so that we have a large enough array. I can hard-code it to 8 if you want.

We'll never have >8 sets in practice,. since if everything is bindless, there will be no packed descriptor set.

Reply:
Defining VKD3D_MAX_DESCRIPTORS to 8 is probably fine.

Reply:
Done.